File: Second-Largest-Element.py
class Solution:
    def secondLargestElement(self, nums):
        L=nums[0]
        SL=0
        for i in range(len(nums)-1):
            if (nums[i]<nums[i+1]) :
                SL=L
                L=nums[i+1]
                
            if L> nums[i]>SL:
                    SL=nums[i]
            elif L>nums[i+1]>SL:
                SL=nums[i+1]
        if SL==0:
            return '-1'
        return SL  

# class Solution:
#     def getSecondLargest(self, arr):
#         # Code Here
#         L=float('-inf')
#         SL=float('-inf')
#         for x in arr:
#             if x>L:
#                 SL=L 
#                 L=x
                
#             elif L>x >SL:
#                     SL=x
            
#         return '-1' if SL==float('-inf') else SL
n=[8, 8, 7, 6, 5]
print(Solution().largestElement(n))

File: Second-Largest-Element.js
// class Solution {
//     secondLargestElement(nums) {
//         let L=nums[0]
//         let SL=0 
//         for(let i=0; i<nums.length-1;i++){
//             if (nums[i]<nums[i+1]){
//                 SL=L
//                 L=nums[i+1]
//             }
//             if (L>nums[i]>SL){
//                 SL=nums[i]
//             }
//         }
//         if (SL===0){
//             return '-1'
//         }
//         else if (L>nums[i+1]>SL){ 
//             SL=nums[i+1]
//         }
//         return SL
//     }
// }
class Solution {
    getSecondLargest(arr) {
        // code her
        let L=-Infinity 
        let SL=-Infinity  
        for (let i=0; i<arr.length ; i++){
            if (arr[i]> L){
                SL=L 
                L=arr[i]
            }    
            else if (L>arr[i]  && arr[i]>SL){ 
                SL=arr[i]
            }
        }
        
        return SL=== -Infinity ? "-1" : SL
    }
}
let n=[7, 7, 2, 2, 10, 10, 10]
console.log(new Solution().getSecondLargest(n))
